drivers/storage/portworx/portworx.go

```diff
@@ -1016,8 +1072,21 @@ func autoUpdateTelemetry(cluster *corev1.StorageCluster) bool {
 		cluster.Spec.Monitoring.Telemetry.Image == ""
 }
 
+func autoUpdateTelemetryProxy(cluster *corev1.StorageCluster) bool {
+	return pxutil.IsTelemetryEnabled(cluster.Spec) &&
+		pxutil.IsCCMGoSupported(pxutil.GetPortworxVersion(cluster))
+}
+
 func autoUpdateMetricsCollector(cluster *corev1.StorageCluster) bool {
-	return pxutil.IsTelemetryEnabled(cluster.Spec)
+	return pxutil.IsTelemetryEnabled(cluster.Spec) &&
+		!pxutil.RunCCMGo(cluster)
```

Comment:
Why can't we just use `pxutil.IsCCMGoSupported` here?

Reply:
addressed offline.